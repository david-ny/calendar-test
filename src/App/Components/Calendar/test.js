import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Calendar from './index';
import { configTestData } from '../../test/testData';

const props = configTestData;

describe('Calendar', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Calendar {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <Calendar {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
