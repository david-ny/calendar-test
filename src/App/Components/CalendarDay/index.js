import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import business from 'moment-business';

const CalendarDay = ({
  dayNum,
  dateString,
  locale,
  prev,
  next,
}) => {
  moment.locale(locale);
  const day = (`0${dayNum}`).slice(-2);
  let monthFromString = parseInt(dateString.slice(5, 7), 10);
  monthFromString = prev ? monthFromString - 1 : monthFromString;
  monthFromString = next ? monthFromString + 1 : monthFromString;
  const month = (`0${monthFromString}`).slice(-2);
  const currentDateString =
    `${dateString.slice(0, 4)}-${month}-${day}`;
  const isWeekDay = business.isWeekDay(moment(currentDateString));
  const color = isWeekDay ? 'gray' : 'red';
  const opacity = (prev || next) ? 0.3 : 1;
  const today = currentDateString === dateString ? 'today' : '';
  return (
    <button
      className={today}
      style={{
         flexBasis: '14.2857%',
         maxWidth: '14.2857%',
         overflow: 'hidden',
         color,
       }}
      type="button"
    >
      <time
        style={{
           opacity,
         }}
        dateTime={moment(currentDateString).format('YYYY-MM-DD HH:MM')}
      >
        {dayNum}
      </time>
    </button>
  );
};

CalendarDay.propTypes = {
  dayNum: PropTypes.number.isRequired,
  dateString: PropTypes.string.isRequired,
  locale: PropTypes.string.isRequired,
  prev: PropTypes.bool,
  next: PropTypes.bool,
};
CalendarDay.defaultProps = {
  prev: false,
  next: false,
};

export default CalendarDay;
