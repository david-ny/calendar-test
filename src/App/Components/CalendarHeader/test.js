import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CalendarHeader from './index';
import { configTestData } from '../../test/testData';

const props = configTestData;

describe('CalendarHeader', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CalendarHeader {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <CalendarHeader {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
