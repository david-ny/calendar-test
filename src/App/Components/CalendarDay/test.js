import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CalendarDay from './index';
import { CalendarDayTestData } from '../../test/testData';

const props = CalendarDayTestData;

describe('CalendarDay', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CalendarDay {...props} />, div);
  });
  test('has a valid snapshot', () => {
    const component = renderer.create(
      <CalendarDay {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
