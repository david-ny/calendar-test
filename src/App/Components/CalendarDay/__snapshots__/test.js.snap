// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay has a valid snapshot 1`] = `
<button
  className=""
  style={
    Object {
      "color": "gray",
      "flexBasis": "14.2857%",
      "maxWidth": "14.2857%",
      "overflow": "hidden",
    }
  }
  type="button"
>
  <time
    dateTime="2018-05-25 00:05"
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    25
  </time>
</button>
`;
